__tests__/sources.ts

```diff
@@ -952,6 +952,55 @@ describe('mutation editSquad', () => {
       'FORBIDDEN',
     );
   });
+
+  it('should throw error when null is sent to memberPostingRole', async () => {
+    loggedUser = '1';
+    return testMutationErrorCode(
+      client,
+      {
+        mutation: MUTATION,
+        variables: { ...variables, memberPostingRole: null },
+      },
+      'GRAPHQL_VALIDATION_FAILED',
+    );
+  });
+
+  it('should edit squad memberPostingRank', async () => {
+    loggedUser = '1';
+    const res = await client.mutate(MUTATION, {
+      variables: {
+        ...variables,
+        memberPostingRole: SourceMemberRoles.Moderator,
+      },
+    });
+    expect(res.errors).toBeFalsy();
+    const editSource = await con
+      .getRepository(SquadSource)
+      .findOneBy({ id: variables.sourceId });
+    expect(editSource?.memberPostingRank).toEqual(
+      sourceRoleRank[SourceMemberRoles.Moderator],
+    );
+  });
+
+  it('should leave squad memberPostingRank unchanged if not sent during edit', async () => {
```

Comment:
Maybe also add the following scenario:

Role is set to owner only
Update only modifies "title"
Should not change role and should not throw error